libpod/container_internal_linux.go

```diff
@@ -391,18 +391,46 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 
 		overlayFlag := false
+		upperDir := ""
+		workDir := ""
 		for _, o := range namedVol.Options {
 			if o == "O" {
 				overlayFlag = true
 			}
+			if overlayFlag && strings.Contains(o, "upperdir") {
+				splitOpt := strings.SplitN(o, "=", 2)
+				if len(splitOpt) > 1 {
+					upperDir = splitOpt[1]
```

Comment:
Should you check for duplicates?

if upperDir != "" {
        return ERROR
}

Reply:
We ignore in underlying library if upperdir is empty. But I agree user should get early error so added here.